### DISCUSSION TITLE
Welcome to Ryzen-5800x3d-linux-undervolting Discussions!

### DISCUSSION THREAD

**Discussion by svenlange2** [svenlange2/Ryzen-5800x3d-linux-undervolting#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by gahabana** [comment]
hi, i do have 5900x - script prints strange values (-l) option - most 0 and few large ints.
I assume 5900x needs to read different bits from the registers - do you know how to 'improve' it so it works on 5900x or at least point me to source of info so i can try to update ?
tx !

**Reply by svenlange2** [reply]
Your issue is (which i cannot confirm as i dont have the cpu) that the 5800x3d is a single CCX (core complex) chip while 5900x is a 2 CCX design. This means that for the first 6 cores you probably see the zeroes (and you probably can write them too). Other two are garbade as they belong to the other CCX and this has probably other register addresses. Which one - I dont know. One place to look into would be the actual CPU programming manual (google). I actually read it a lot to figure out what to do. But again - I dealt with 5800X3D.

**Comment by toscdesign** [comment]
Thank you so much for this tool 👍

It works perfectly fine with my Ryzen 5800X3D and old ASUS Prime X370 Pro on EndeavourOS

**Comment by kimnux12** [comment]
Thank you very much!!!!  I use asus b450m-a motherboard which does not have pbo2 option. 
It works perfectly. 
But, is there an idea about automation? 
I make a batch file, and typing it at every login.  It's little tiresome. 
Sorry for my poor english.

**Reply by Termuellinator** [reply]
i just created a systemd service for it:
```
[Unit]
Description=Set the voltage offset to -30 for all cores

[Service]
ExecStart=python <path to ruv.py> -c 8 -o -30

[Install]
WantedBy=multi-user.target

```

**Reply by kimnux12** [reply]
wow thank you. I will try it.
